lib/model/forum/ForumDradenMeldingModel.class.php

```diff
@@ -0,0 +1,192 @@
+		$ontvangerNaam = $ontvanger->getNaam('civitas');
+		$values = array(
+			'NAAM' => $ontvangerNaam,
+			'AUTEUR' => $auteur->getNaam('civitas'),
+			'POSTLINK' => $post->getLink(true),
+			'TITEL' => $draad->titel,
+			'TEKST' => str_replace('\r\n', "\n", $post->tekst),
+		);
+
+		// Stel huidig UID in op ontvanger om te voorkomen dat ontvanger privé of andere persoonlijke info te zien krijgt
+		$curUID = $_SESSION['_uid'];
+		$_SESSION['_uid'] = $ontvanger->uid;
+
+		// Verzend mail
+		$mail = new Mail(array($ontvanger->getPrimaryEmail() => $ontvangerNaam), 'C.S.R. Forum: nieuwe reactie op ' . $draad->titel, $bericht);
+		$mail->setPlaceholders($values);
+		$mail->setLightBB();
+		$mail->send();
+
+		// Zet UID terug in sessie
+		$_SESSION['_uid'] = $curUID;
+	}
+
+	/**
+	 * Stuurt meldingen van nieuw bericht naar leden met meldingsniveau op altijd
+	 *
+	 * @param ForumPost $post
+	 */
+	public function stuurMeldingenNaarVolgers(ForumPost $post) {
+		$auteur = ProfielModel::get($post->uid);
+		$draad = $post->getForumDraad();
+
+		// Laad meldingsbericht in
+		$bericht = file_get_contents(SMARTY_TEMPLATE_DIR . 'mail/forumaltijdmelding.mail');
+		foreach ($this->getAltijdMeldingVoorDraad($draad) as $volger) {
+			$volger = ProfielModel::get($volger->uid);
+
+			// Stuur geen meldingen als lid niet gevonden is of lid de auteur
+			if (!$volger || $volger->uid === $post->uid) {
+				continue;
+			}
+			$this->stuurMelding($volger, $auteur, $post, $draad, $bericht);
+		}
+	}
+
+	/**
+	 * Zoek genoemde leden in gegeven bericht
+	 *
+	 * @param string $bericht
+	 * @return string[]
+	 */
+	public function zoekGenoemdeLeden($bericht) {
+		$regex = "/\[(?:lid|citaat)=?\s*]?\s*([[:alnum:]]+)\s*(?:\]|\[)/";
+		preg_match_all($regex, $bericht, $leden);
+
+		return $leden[1];
+	}
+
+	/**
+	 * Stuurt meldingen van nieuw bericht naar leden die genoemd / geciteerd worden in bericht
+	 *
+	 * @param ForumPost $post
+	 */
+	public function stuurMeldingenNaarGenoemden(ForumPost $post) {
+		$auteur = ProfielModel::get($post->uid);
+		$draad = $post->getForumDraad();
+
+		// Laad meldingsbericht in
+		$bericht = file_get_contents(SMARTY_TEMPLATE_DIR . 'mail/forumvermeldingmelding.mail');
+		$genoemden = $this->zoekGenoemdeLeden($post->tekst);
+		foreach ($genoemden as $uid) {
+			$genoemde = ProfielModel::get($uid);
+
+			// Stuur geen meldingen als lid niet gevonden is, lid de auteur is of als lid geen meldingen wil voor draadje
+			// Met laatste voorwaarde worden ook leden afgevangen die sowieso al een melding zouden ontvangen
+			if (!$genoemde || $genoemde->uid === $post->uid || $this->getNiveauVoorLid($draad, $genoemde->uid) !== ForumDraadMeldingNiveau::VERMELDING) {
+				continue;
+			}
+
+			// Controleer of lid bij draad mag, stel hiervoor tijdelijk de ingelogde gebruiker in op gegeven lid
+			$curUid = $_SESSION['_uid'];
+			$_SESSION['_uid'] = $genoemde->uid;
+			$magMeldingKrijgen = $draad->magMeldingKrijgen();
```

Comment:
Ik zou aan magMeldingKrijgen in ieder geval een (optioneel) argument met het betreffende account meegeven. Dan is het hier niet nodig.

Reply:
In Loginmodel zit dat volgens mij al wel, maar die check in forumdraad is best wel ingewikkeld

Reply:
Zo te [zien](https://github.com/csrdelft/csrdelft.nl/blob/c76cd7e0a59ca8f4f8ac08d618b646789ac63c92/lib/model/security/LoginModel.class.php#L51) zit er alleen een modus in voor de CLI, voor de rest komt het huidige uid altijd uit de sessie.